### PR TITLE
EDGPATRON-33: Add patron notes to requests (EDGEPATRON)

### PR DESCRIPTION
**Pull request by oleksandr-yatsenko1**
https://issues.folio.org/browse/EDGPATRON-33
EDGPATRON-33: Add patron notes to requests (EDGEPATRON)

## Approach
* Added support of patronComments functionality.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by marcjohnson-kint — COMMENTED** [review]
Are there tests for placing a request?

**Comment by oleksandr-yatsenko1** [comment]
> Are there tests for placing a request?

Good point, will check it, thanks.

File: src/test/java/org/folio/edge/patron/MainVerticleTest.java [hunk 1]
```diff
@@ -590,7 +592,7 @@ public void testPlaceInstanceHoldInstanceNotFound(TestContext context) throws Ex
     logger.info("=== Test place instance hold w/ instance not found ===");
 
     Hold hold = PatronMockOkapi.getHold(PatronMockOkapi.instanceId_notFound);
-    int expectedStatusCode = 501;
+    int expectedStatusCode = 404;
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/edge/patron/MainVerticleTest.java:595]: Why has the behaviour of this existing test changed? How is it related to this change, which is to add an extra property to the incoming request?

Comment (oleksandr-yatsenko1) [reply src/test/java/org/folio/edge/patron/MainVerticleTest.java:595]: The point here that there was no handling of "post hold for patron on instance level" processing before. I determined that previously server was returning only an error code 501:

  public void placeInstanceHoldHandler(RoutingContext ctx) {
    ctx.response()
      .setStatusCode(501)
      .end();
  }

Whereas the same handler of patron hold creation on items level has some processing with different kind of error responses. I suspect that was uncompleted functionality of route stubbing. That why some of tests from MainVerticleTest contained odd cases like "testPlaceInstanceHoldSuccess" which despite implying success of operation expect a response code 501. I added implementation for instance level handling and as a result changed expected error code for a value returned by the placeItemHoldHandler.

File: src/main/java/org/folio/edge/patron/Constants.java [hunk 2]
```diff
@@ -35,6 +35,8 @@
   public static final String FIELD_CANCELLATION_ADDITIONAL_INFO = "cancellationAdditionalInformation";
   public static final String FIELD_CANCELED_DATE = "canceledDate";
 
+  public static final String FIELD_PATRON_COMMENTS = "patronComments";
+
```
[the remark this answers is not in the document]
Comment (bohdan-suprun) [reply src/main/java/org/folio/edge/patron/Constants.java]: I have removed it, it is unused.

**Review by bohdan-suprun — APPROVED** [review]

File: src/main/java/org/folio/edge/patron/Constants.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleksandr-yatsenko1) [reply src/main/java/org/folio/edge/patron/Constants.java]: Yes, as it turned out it unneeded here. Thank you Bohdan.

**Review by marcjohnson-kint — APPROVED** [review]
Thank you for improving the existing tests. 

It would've helped me if those changes had been explained in the description of the pull request or (my preference) had been submitted as a separate pull request so the changes to make the tests more representation of the behaviour before the change was separate to the change itself

**Pull request merged**